maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/NodeHelper.java

```diff
@@ -32,19 +36,38 @@ public class NodeHelper {
 	
 	private final BuildPluginManager pluginManager;
 
-	private Path tempDir;
-	
-	public NodeHelper(MavenProject mavenProject,
-			MavenSession mavenSession,
-			BuildPluginManager pluginManager) {
+	private final Path tempDir;
+
+	private static final Logger logger = LoggerFactory.getLogger(NodeHelper.class);
+
+	public Path getTempDir() {
+		return tempDir;
+	}
+
+	public NodeHelper(MavenProject mavenProject, MavenSession mavenSession, BuildPluginManager pluginManager) {
 		this.mavenProject = mavenProject;
 		this.mavenSession = mavenSession;
 		this.pluginManager = pluginManager;
 		try {
-			this.tempDir = Files.createTempDirectory("openmrs-sdk-node");
-			this.tempDir.toFile().deleteOnExit();
+			Properties sdkProperties = getSdkProperties();
+			boolean isRemoveCacheAfterSetup = Boolean.parseBoolean(sdkProperties.getProperty("removeCacheAfterSetup"));
+			String cacheName = "openmrs-sdk-node";
+			if (!isRemoveCacheAfterSetup) {
+				logger.info("OpenMRS SDK will reuse the NPM cache. Run openmrs-sdk:setup-sdk -DremoveCacheAfterSetup=true to disable it.");
```

Comment:
Can we be consistent? The description says `reuse npm cache` but the method name says `remove cache`.

Reply:
Should we rename the parameter as "ReuseNpmCache" ??

Reply:
I think logger info is consistent with implementation. If isRemoveCacheAfterSetup is false the NPM will reuse NPM caches without deleting them.

Reply:
How about something along the lines of `reuseNodeCache`?

Reply:
sounds good

Reply:
I changed the parameter name